server/controllers/offer/addOffer.js

```diff
@@ -1,3 +1,28 @@
-module.exports = (req, res) => {
-  res.send(' add new offer');
+const {
+  addOfferDetails,
+  addOfferSkill,
+  addOfferTypes,
+} = require('./../../database/queries/offers/index');
+
+
+module.exports = (req, res, next) => {
+  const {
+    title, position, description, skills, offer_types, member_id,
+  } = req.body;
+  let offerDetails;
+  addOfferDetails(title, position, description, member_id)
+    .then((response) => {
+      skills.map((item) => {
```

Comment:
@AhmedAl-Almi I think you might need to use `Promise.all` to wait for all of the promises returned by addOfferSkill before proceeding:
```
return Promise.all(skills.map((item) => {
  offerDetails = .....
  return addOfferSkill( offerDetails.id, item.id );
}));
```
So now the inside of the map is returning addOfferSkill's promise, which means that the map is returning an array of promises, and all of those are being passed to Promise.all() which will resolve once all of the promises inside that array have all resolved.

I think the same thing would help on line 20, too:
`.then(() => Promise.all(offer_types.map(item => addOfferTypes(offerDetails.id, item.id))))`

This will make sure that your endpoint is waiting for all of the queries to finish, at each stage, before proceeding.  Also, this will make sure that any query that fails will actually be collected and handled by your catch() on line 27.

Reply:
@tomduggan85 ok, good suggestion :+1: